packages/app/src/vis-packs/nexus/hooks.ts

```diff
@@ -62,15 +68,22 @@ export function useAxisMapping(
 }
 
 export function useAuxiliaries(
-  auxDatasets: NumArrayDataset[],
+  auxDatasets: AuxDatasets,
   selection: string | undefined
 ): Auxiliary[] {
   const { attrValuesStore } = useDataContext();
 
-  const auxValues = useDatasetValues(auxDatasets, selection);
+  const auxValues = useDatasetValues(
+    [
+      ...auxDatasets.map((d) => d.signal),
+      ...auxDatasets.map((d) => d.errors).filter(isDefined),
+    ],
+    selection
+  );
 
-  return auxDatasets.map((dataset) => ({
-    label: getDatasetLabel(dataset, attrValuesStore),
-    value: auxValues[dataset.name],
+  return auxDatasets.map(({ signal, errors }) => ({
+    label: getDatasetLabel(signal, attrValuesStore),
+    value: auxValues[signal.name],
+    error: errors ? auxValues[errors.name] : undefined,
```

Comment:
I'm also tempted to suggest pluralising the keys in the object: `values` and `errors`, since they each hold an array of data/error values.

Maybe we'll have to do a pass later to decide once and for all when to pluralise those two terms. Because even though `errors` and `values` on their own need to be pluralised when they hold an array, they don't necessarily need to be when used as [attributive nouns](https://english.stackexchange.com/a/397244):

- `errorArray`: the array that holds the error information
- `errorsArray`: the array that holds multiple error values (to convey variety) or the array that "belongs to" the errors (to convey possession)

I'd argue that we don't really need to convey variety or possession, and therefore that the singular form might in fact be more appropriate.

Reply:
Make sense